### PR TITLE
Perform action for busy, no-answer and failed calls

### PR DESCRIPTION
**Pull request by dwhenry**
This code has been tested and is ready for deployment.

### CODE REVIEW

File: app/models/twilio_redirection.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@ def initialize(location)
       @location = location
     end
 
+    delegate :title, to: :@location
```
Comment (benlovell) [inline app/models/twilio_redirection.rb]: This is delegated by also you provide a concrete example below that returns an empty string. What's the intent?

File: app/views/twilios/show.xml.builder [hunk 2]
```diff
@@ -1,6 +1,6 @@
 xml.instruct!
 xml.Response do
-  xml.Dial do
+  xml.Dial(action: twilio_handle_status_url, method: "GET") do
```
Comment (benlovell) [inline app/views/twilios/show.xml.builder]: Single quotes.

File: spec/controllers/twilios_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+      it 'raise a Bugsnag error' do
+        get :handle_status, params
+        expect(Bugsnag).to have_received(:notify).with(
+          "Invalid number detected for: '#{location.title}' (#{location.phone})"
+        )
+      end
+    end
+
+    context 'call status is busy called with a Called number' do
```
Comment (benlovell) [inline spec/controllers/twilios_controller_spec.rb]: This is kinda hard to parse.

File: app/models/twilio_redirection.rb (hunk 1 above)
Comment (dwhenry) [inline app/models/twilio_redirection.rb]: This was two classes in one file. This is now split into separate files.

File: spec/controllers/twilios_controller_spec.rb (hunk 3 above)
Comment (dwhenry) [inline spec/controllers/twilios_controller_spec.rb]: Hopefully this is easier to read now.

**Comment by dwhenry** [comment]
@benlovell ready to review when you are ready.

**Comment by AndrewVos** [comment]
@dwhenry lgtm
